sourced/ml/transformers/uast2termfreq.py

```diff
@@ -0,0 +1,25 @@
+from collections import namedtuple
+import operator
+
+from sourced.ml.transformers import Transformer
+
+TermFreqRow = namedtuple("TermFreqRow", ["token", "document", "value"])
+
+
+class Uast2TermFreq(Transformer):
+    def __init__(self, extractors, document_column, **kwargs):
+        super().__init__(**kwargs)
+        self.extractors = extractors
+        self.document_column = document_column
+
+    def __call__(self, rows):
+        return rows \
+            .flatMap(self.process_row) \
+            .reduceByKey(operator.add) \
+            .map(lambda x: TermFreqRow(document=x[0][0], token=x[0][1], value=x[1]))
```

Comment:
Would be great to use `Row` or `namedtuple` here too

Reply:
but I use it. What do you mean?

Reply:
I mean `x[0][0]` and friends

Reply:
oh, ok. about it and https://github.com/src-d/ml/pull/160#discussion_r161794558

I can not do here anything more since each operation will override my `namedtuple` (or `pyspark.Row`, what a little bit strange) with usual `tuple`, and I get `AttributeError: 'tuple' object has no attribute 'token'`. I do not find the solution to this problem.

One thing I can do is to use `.toDF()` at the end of `df` and `tf` calculation. In this case it will look better. One problem, we agreed that `Transformer` should return `rdd` or something iterable with tuples.

Reply:
or not... I have one idea. will check.

Reply:
Only like this:

```python
return self.tf \
            .map(lambda x: (x.token, Row(document=x.document, tf=x.value))) \
            .join(df_pruned) \
            .map(lambda x: Row(token=x[0],
                               document=x[1][0].document,
                               value=log_tf_log_idf(df=x[1][1], tf=x[1][0].tf, ndocs=ndocs)))
```

Not sure it is better...

Reply:
All right, if it is WONTFIX, then please comment every index